sdk/examples/data_hash.rs

```diff
@@ -148,7 +148,7 @@ fn user_data_hash_with_sdk_hashing() {
     output_file.write_all(&final_manifest).unwrap();
 
     // make sure the output file is correct
-    let manifest_store = ManifestStore::from_file(&dest).unwrap();
```

Comment:
This shift from `ManifestStore` to `Reader` seems odd. What is the semantic intent here?

Reply:
At first I was going to update the whole code base to use Reader instead of ManifestStore and remove ManifestStore entirely. I changed course and decided to leave the old APIs in for now. This is one place I had updated to Reader. I can change it back for now if you prefer, but it will become Reader eventually.

Reply:
I reverted this for now to be more consistent as an example.